core/chains/evm/txmgr/finalizer.go

```diff
@@ -280,16 +348,290 @@ func (f *evmFinalizer) batchCheckReceiptHashesOnchain(ctx context.Context, block
 	return finalizedReceipts
 }
 
-// Build list of transaction IDs
-func (f *evmFinalizer) buildReceiptIdList(finalizedReceipts []Receipt) []int64 {
-	receiptIds := make([]int64, len(finalizedReceipts))
+func (f *evmFinalizer) FetchAndStoreReceipts(ctx context.Context, head, latestFinalizedHead *evmtypes.Head) error {
+	attempts, err := f.txStore.FindAttemptsRequiringReceiptFetch(ctx, f.chainID)
```

Comment:
Is there a way to separate an overflowed from a fatal transaction? Although they are both classified as fatal, the former has a receipt, meaning this check will eventually stop. For the latter, seems like we'll try to fetch receipts indefinitely.

Reply:
I believe that's implicitly separated by us returning attempts. Other transactions marked as fatal in the Broadcaster delete their attempts so this query wouldn't pick those up.